manim/mobject/geometry/polygram.py

```diff
@@ -65,17 +65,10 @@ def construct(self):
     """
 
     def __init__(self, *vertex_groups: Iterable[Sequence[float]], color=BLUE, **kwargs):
+        self._initial_vertex_groups = np.array(vertex_groups)
```

Comment:
Coming out of hiding to say that I don't think I much like the changes to `Polygram`.

It may be better to pass a `closed_loop=True` parameter to the constructor of `Polygram` to determine whether `first_vertex` should be tacked onto the end of the vertices. Then `Polyline` could just pass up `closed_loop=False`. I also have always had a general disdain for the `generate_points` pattern, requiring initial data to be stored for when it gets called instead of just using the information you already have in the constructor.

With the `vertex_group_shape` attribute, I do not like its addition, especially as it will not update when the mobject's points update. So if a square transforms into a triangle, then `get_vertex_groups` will give incorrect results. I am not sure what else could be done to support a vertex group that isn't a closed loop however.

I feel the best approach is to have `Polyline` unrelated to `Polygram` altogether, as while they may share certain properties and may make sense to have the same utilities (i.e. `round_corners`), I believe mathematically speaking a polygram only consists of closed loops, while polylines do not have to be closed loops. It however does not make sense to me to have `Polygram` inherit from `Polyline`, as I believe a polyline can only make a poly**gon** if it is a closed loop. Therefore I think `Polyline` should be separated from the `Polygram` class and perhaps have shared utilities for e.g. rounding their corners. This would align with their mathematical definitions (to my understanding), and would I think make more sense in the code.

Reply:
I would have touched `Polygram` sooner had I known that would bring you out of hiding! 😄 

On a more serious note, I actually agree. The reason I rewrote it like this was because I wanted to avoid code duplication and got *inspired* by the situation in 3b1b/manim where `Polyline` inherits from `Polygon`. I'm happy to change `Polygram` back to the old implementation.

As for `Polyline`, I could see it fit well as a parent class of `Line` (or as an ad-hoc implementation that literally just calls `VMobject.set_points_as_corners`); your note on shared utilities with other classes has been at the back of my mind for a while as well and eventually I'd like to implement something like this perhaps with a mixin-class sort of approach (which I feel is out of scope for this PR, as it is (too) large and barely reviewable as it is).

Reply:
> I would have touched Polygram sooner had I known that would bring you out of hiding! :smile:

It's one of the more visible and concrete parts of manim that I worked on, I'm a tad protective of it, and this PR piqued my interest :P

> The reason I rewrote it like this was because I [...] got *inspired* by the situation in 3b1b/manim

~~Well there's the issue.~~ Having 3b1b/manim code "infect" code I wrote may also have struck a sore spot for me.

> As for Polyline, I could see it fit well as a parent class of Line

This seems like a good fit to me. Any shared utilities could be later added, perhaps through a common base class, something like `_PolyBase`, which doesn't need to have any semantic or user-facing meaning.

Reply:
4a3cfcb1 restores `polgram` to its previous state, and 547db1c1 introduces a admittedly low-effort compromise (I actually don't think a dedicated `Polyline` mobject is really necessary at this point, and `Line` has stuff like the processing of `buff` etc, which I feel should be moved to a hypothetical `Polyline` parent class -- but I don't really want to implement all that with this PR).

Reply:
Seems good to me